go.mod

```diff
@@ -1,10 +1,12 @@
 module github.com/google/wire
 
-go 1.12
+go 1.23
```

Comment:
Can we bump to 1.22 to support the most recent 2 Go versions? Like https://go.dev/doc/devel/release#policy

Reply:
Well 1.24 released so can we set to 1.23 now?

Reply:
Well 1.24 released so can we set to 1.23 now?

Reply:
FWIW, I thanks to go 1.24's latest "tool" feature, we no longer need any new versions… we can just invoke it as a tool, which will recompile it every time.